presto-main/src/test/java/com/facebook/presto/sql/planner/optimizations/TestSimplifyPredicates.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.planner.optimizations;
+
+import com.facebook.presto.sql.parser.SqlParser;
+import com.facebook.presto.sql.tree.Expression;
+import com.facebook.presto.sql.tree.ExpressionRewriter;
+import com.facebook.presto.sql.tree.ExpressionTreeRewriter;
+import com.facebook.presto.sql.tree.LogicalBinaryExpression;
+import org.testng.annotations.Test;
+
+import java.util.List;
+
+import static com.facebook.presto.sql.ExpressionUtils.binaryExpression;
+import static com.facebook.presto.sql.ExpressionUtils.extractPredicates;
+import static com.facebook.presto.sql.planner.optimizations.SimplifyPredicates.simplifyPredicates;
+import static java.util.stream.Collectors.toList;
+import static org.testng.Assert.assertEquals;
+
+public class TestSimplifyPredicates
+{
+    private static final SqlParser SQL_PARSER = new SqlParser();
+
+    @Test
+    public void testExtractsCommonPredicate()
+    {
+        assertSimplifies("X AND X", "X");
+        assertSimplifies("X OR X", "X");
+
+        assertSimplifies("(X AND Y AND V) OR (X AND Y AND Z)", "(X AND Y) AND (V OR Z)");
+        assertSimplifies("(X OR Y OR V) AND (X OR Y OR Z)", "(X OR Y) OR (V AND Z)");
+
+        assertSimplifies("((X OR V) AND V) OR ((X OR V) AND V)", "(X OR V) AND V");
+        assertSimplifies("((X OR V) AND X) OR ((X OR V) AND V)", "X OR V");
```

Comment:
Can you add a tests with nested common parts like
`(x and ((y and z) or (y and v)) or (x and y)`
To see how `y` will be handled.